File: 1480_map.ts
function runningSum(nums: number[]): number[] {
    let sum = 0
    return nums.map(cur => sum += cur)
}

console.log(runningSum([1,2,3,4]))
console.log(runningSum([1,1,1,1,1]))
console.log(runningSum([3,1,2,10,1]))

File: 1480_map.go
func mapInt(x []int, f func(int) int) []int {
    r := make([]int, len(x))
    for i, e := range x { r[i] = f(e) }
    return r
}

func runningSum(nums []int) []int {
  sum := 0
  return mapInt(nums, func(num int) int { sum += num; return sum })
}
